histomicstk/annotations_and_masks/annotations_to_masks_handler.py

```diff
@@ -834,23 +943,34 @@ def get_all_rois_from_slide(
     # get bounding box information for all annotations
     element_infos = get_bboxes_from_slide_annotations(slide_annotations)
 
-    # get idx of all 'special' roi annotations
-    idxs_for_all_rois = _get_idxs_for_all_rois(
-        GTCodes=GTCodes_df, element_infos=element_infos)
+    # define roi_getter, which yields one roi at a time
+    if max_roiside is None:
+        roig = _roi_getter_asis(
+            gc=gc, slide_id=slide_id, GTCodes_dict=GTCodes_dict,
+            slide_annotations=slide_annotations, element_infos=element_infos,
+            get_kwargs=kvp, monitor=monitorPrefix, verbose=verbose,
+        )
+    else:
+        roig = _roi_getter_tiled(
+            gc=gc, slide_id=slide_id, GTCodes_dict=GTCodes_dict,
+            slide_annotations=slide_annotations, element_infos=element_infos,
+            sf=sf, max_roiside=max_roiside,
+            get_kwargs=kvp, monitor=monitorPrefix, verbose=verbose,
+        )
 
     savenames = []
 
-    for roino, idx_for_roi in enumerate(idxs_for_all_rois):
-
-        roicountStr = "%s: roi %d of %d" % (
-            monitorPrefix, roino + 1, len(idxs_for_all_rois))
+    while True:
```

Comment:
Instead of `while True` through the `StopIteration` line, this could be reduce to `for roi_out in roig:`

Reply:
@manthey Thanks! I've incorporated your correction.